scipy/stats/_qmc.py

```diff
@@ -1595,6 +1596,302 @@ def fast_forward(self, n: IntNumber) -> Sobol:
         return self
 
 
+class PoissonDisk(QMCEngine):
+    """Poisson disk sampling.
+
+    Parameters
+    ----------
+    d : int
+        Dimension of the parameter space.
+    radius : float
+        Minimal distance to keep between points when sampling new candidates.
+    hypersphere : {"volume", "surface"}, optional
+        Sampling strategy to generate potential candidates to be added in the
+        final sample. Default is "volume".
+
+        * ``volume``: original Bridson algorithm as described in [1]_.
+          new candidates are sampled within the hypersphere.
+        * ``surface``: only sample the surface of the hypersphere.
+
+    candidates : int
```

Comment:
What do you mean by more directly?

Reply:
Often I want to put, say, 15 points into an area, ensuring they don't touch. It does not look as though you can do that with the current implementation.

Reply:
You can, it's just `random(15)`. The issue is that it can return less than 15 points. It would be hard to generalize and ensure we can find exactly 15 points. We would need to change the radius to make this happen. But then we would not be able to potentially add more samples or it would do so by reducing the radius. IMO not clear. Can make sense, but definitely a deviation from the base Poisson sampling.

This parameter is just to say that at each iterations how many points around a given point we are sampling. Then we look at these candidates one by one and see if they are far enough from existing points. If yes we add them and to the sample and the pool of centre points to use in next iterations.

Reply:
`random(15)`? I thought the argument is an int?

Reply:
I meant `PoissonDisk(...).random(15)`

Reply:
Ah! That makes sense. If you happen to have enough points in the poisson disk, but I suppose you can write a loop to keep trying.

Reply:
This is a bit what the method `full_space` does. Try to add as many points as possible.